### ISSUE TITLE
IcingaCheckUsedPartitionSpace - UNKNOWN on no disk space

### ISSUE THREAD

**Issue by davekempe** [Icinga/icinga-powershell-plugins#200]
Hi,
had an incident on the weekend where a disk filled up very quickly, and the Icinga plugin threw an UNKNOWN, not a CRITICAL.

Of course, in this instance, of all the things to go critical, this plugin should. I'm not sure why UsedPartitionSpace requires space to run, but let me know if there is anything else I can provide.

Windows Server 2019, fully patched, only a couple of weeks old.

This is the check output:

```
UNKNOWN
--
[UNKNOWN] Icinga Exception: There is not enough space on the disk.CheckCommand: Invoke-IcingaCheckUsedPartitionSpace Arguments: -Include System.Object[] -Exclude System.Object[] StackTrace: at Get-IcingaDiskAttributes, C:\Program Files\WindowsPowerShell\Modules\icinga-powershell-plugins\1.5.0\provider\disks\Get-IcingaDiskAttributes.psm1: line 37 at Get-IcingaPhysicalDiskInfo, C:\Program Files\WindowsPowerShell\Modules\icinga-powershell-plugins\1.5.0\provider\disks\Get-IcingaPhysicalDiskInfo.psm1: line 49 at Invoke-IcingaCheckUsedPartitionSpace, C:\Program Files\WindowsPowerShell\Modules\icinga-powershell-plugins\1.5.0\plugins\Invoke-IcingaCheckUsedPartitionSpace.psm1: line 90 at Exit-IcingaExecutePlugin, C:\Program Files\WindowsPowerShell\Modules\icinga-powershell-framework\1.5.0\lib\icinga\plugin\Exit-IcingaExecutePlugin.psm1: line 17 at , : line 1
```

**Comment by LordHepipud** [comment]
This is an interesting issue which I have to look on. The only assumption I could make, is that the loading if certain data including fetching of counters and other objects requires free disk space.

I will try to re-produce this and work on a fix.

**Comment by davekempe** [comment]
Please let me know if there is anything else I can provide for you. Happy to have a session to share my screen and show you the environment even if that helps.

**Comment by LordHepipud** [comment]
Yes I could re-produce the issue and still figuring out, on how I can ensure we receive an critical in this instance.
Once I have some more input, I will create a PR for testing.

**Comment by LordHepipud** [comment]
Hello,

I just created the linked PR, containing an improvement for this behaviour. Could you please test this PR?
You will require the latest snapshot/master from the icinga-powrshell-framework as well for this.

In case the disk has no space left, the message in the future on the disk plugin will look like this:

```powershell
icinga { Invoke-IcingaCheckUsedPartitionSpace -Verbosity 2 }

[CRITICAL] Failed to process disk related checks. Based on the error your local Windows disk partition has no space left
```

During runtime on this position there are no additional information left, but this is only happening in case the main disk partition (usually C:) is having no space left.

If everything works as expected, I will merge the PR and it will be released with 1.6.0.

**Comment by LordHepipud** [comment]
Have you been able to test the plugin and check, if the issue is resolved?
